pkg/operator/operator.go

```diff
@@ -249,6 +294,12 @@ func New(logger logr.Logger, clientConfig *rest.Config, registry prometheus.Regi
 		return nil, errors.Wrap(err, "create client")
 	}
 
+	if !opts.TargetPollDisabled {
```

Comment:
I don't think it makes sense to store the registry since we won't be adding new metrics at runtime, but I think it makes sense to have the registry as a parameter in the `Run`. In fact, we weren't even using it in the constructor previously (although we took it in) until I added the target status controller. Thoughts?

Reply:
Ah nice - agreed!